File: internal/domain/command.go
package domain

import "time"

type CommandAction string

const (
	CmdLockDevice  CommandAction = "lock_device"
	CmdWipeDevice  CommandAction = "wipe_device"
	CmdCapturePhoto CommandAction = "capture_photo"
	CmdTriggerAlarm CommandAction = "trigger_alarm"
)

type CommandStatus string

const (
	CmdPending      CommandStatus = "pending"
	CmdSent         CommandStatus = "sent"
	CmdReceived     CommandStatus = "received"
	CmdExecuted     CommandStatus = "executed"
	CmdFailed       CommandStatus = "failed"
	CmdTimedOut     CommandStatus = "timed_out"
)

type Command struct {
	ID              string        `json:"id"`
	DeviceID        string        `json:"device_id"`
	Action          CommandAction `json:"action"`
	Params          string        `json:"params,omitempty"`
	Status          CommandStatus `json:"status"`
	CreatedAt       time.Time     `json:"created_at"`
	SentAt          *time.Time    `json:"sent_at,omitempty"`
	AcknowledgedAt  *time.Time    `json:"acknowledged_at,omitempty"`
	CompletedAt     *time.Time    `json:"completed_at,omitempty"`
	Error           *string       `json:"error,omitempty"`
}

File: app/lib/models/command.dart
class RemoteCommand {
  final String cmdId;
  final String action;
  final Map<String, dynamic>? params;
  final DateTime timestamp;
  final String status;

  RemoteCommand({
    required this.cmdId,
    required this.action,
    this.params,
    required this.timestamp,
    this.status = 'pending',
  });

  static const String lockDevice = 'lock_device';
  static const String wipeDevice = 'wipe_device';
  static const String capturePhoto = 'capture_photo';
  static const String triggerAlarm = 'trigger_alarm';

  Map<String, dynamic> toJson() => {
        'type': 'command',
        'cmd_id': cmdId,
        'action': action,
        'params': params,
        'ts': timestamp.toIso8601String(),
      };

  Map<String, dynamic> ackJson() => {
        'type': 'ack',
        'cmd_id': cmdId,
        'status': 'received',
        'ts': DateTime.now().toIso8601String(),
      };

  Map<String, dynamic> resultJson(String resultStatus, {String? error}) => {
        'type': 'result',
        'cmd_id': cmdId,
        'status': resultStatus,
        'error': error,
        'ts': DateTime.now().toIso8601String(),
      };

  factory RemoteCommand.fromJson(Map<String, dynamic> json) {
    return RemoteCommand(
      cmdId: json['cmd_id'] as String,
      action: json['action'] as String,
      params: json['params'] as Map<String, dynamic>?,
      timestamp: json['ts'] != null
          ? DateTime.parse(json['ts'])
          : DateTime.now(),
      status: json['status'] as String? ?? 'pending',
    );
  }
}
